e2e_tests/tests/per_provider/normal_tests/create_destroy_key.rs

```diff
@@ -37,53 +46,67 @@ fn destroy_without_create() {
     assert_eq!(status, ResponseStatus::PsaErrorDoesNotExist);
 }
 
+#[cfg(not(feature = "cryptoauthlib-provider"))]
 #[test]
-fn create_destroy_and_operation() {
+fn create_destroy_and_operation() -> Result<()> {
     let mut client = TestClient::new();
     let hash = vec![0xDE; 32];
     let key_name = String::from("create_destroy_and_operation");
 
-    client.generate_rsa_sign_key(key_name.clone()).unwrap();
+    client.generate_rsa_sign_key(key_name.clone())?;
 
-    client.destroy_key(key_name.clone()).unwrap();
+    client.destroy_key(key_name.clone())?;
 
     let status = client
         .sign_with_rsa_sha256(key_name, hash)
         .expect_err("The key used by this operation should have been deleted.");
     assert_eq!(status, ResponseStatus::PsaErrorDoesNotExist);
+
+    Ok(())
 }
 
 #[test]
-fn create_destroy_twice() {
+fn create_destroy_twice() -> Result<()> {
     let mut client = TestClient::new();
     let key_name = String::from("create_destroy_twice_1");
     let key_name_2 = String::from("create_destroy_twice_2");
 
-    client.generate_rsa_sign_key(key_name.clone()).unwrap();
-    client.generate_rsa_sign_key(key_name_2.clone()).unwrap();
-
-    client.destroy_key(key_name).unwrap();
-    client.destroy_key(key_name_2).unwrap();
+    #[cfg(not(feature = "cryptoauthlib-provider"))]
+    {
+        client.generate_rsa_sign_key(key_name.clone())?;
+        client.generate_rsa_sign_key(key_name_2.clone())?;
+    }
+    #[cfg(feature = "cryptoauthlib-provider")]
+    {
+        client.generate_ecc_key_pair_secpr1_ecdsa_sha256(key_name.clone())?;
+        client.generate_ecc_key_pair_secpr1_ecdsa_sha256(key_name_2.clone())?;
+    }
+
+    client.destroy_key(key_name)?;
+    client.destroy_key(key_name_2)
 }
 
+#[cfg(not(feature = "cryptoauthlib-provider"))]
 #[test]
-fn generate_public_rsa_check_modulus() {
+fn generate_public_rsa_check_modulus() -> Result<()> {
     // As stated in the operation page, the public exponent of RSA key pair should be 65537
     // (0x010001).
     let mut client = TestClient::new();
     let key_name = String::from("generate_public_rsa_check_modulus");
-    client.generate_rsa_sign_key(key_name.clone()).unwrap();
-    let public_key = client.export_public_key(key_name).unwrap();
+    client.generate_rsa_sign_key(key_name.clone())?;
+    let public_key = client.export_public_key(key_name)?;
 
     let public_key: RSAPublicKey = picky_asn1_der::from_bytes(&public_key).unwrap();
     assert_eq!(
         public_key.public_exponent.as_unsigned_bytes_be(),
         [0x01, 0x00, 0x01]
     );
+    Ok(())
 }
 
+#[cfg(not(feature = "cryptoauthlib-provider"))]
```

Comment:
Could you do it through:
```rust
    if !client.is_operation_supported(Opcode::PsaImportKey) {
        return Ok(());
    }
```
?
In case the operation is added in the future

Reply:
I could, but there is more to be done - ATECC does not support RSA, it supports ECC only. I plan to leave this for the time being.